### ISSUE TITLE
Make Documentation optional to avoid a warning

### ISSUE THREAD

**Issue by ahoarau** [Simple-Robotics/proxsuite#175]
Hello,
I'm trying to build proxsuite on a system without doxygen, I get:
```
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
CMake Warning at cmake-module/doxygen.cmake:494 (message):
  Failed to find Doxygen, documentation will not be generated.
Call Stack (most recent call first):
  cmake-module/post-project.cmake:59 (_setup_project_documentation)
  CMakeLists.txt:45 (project)
  ```

### Proposed solution
Add a `BUILD_DOCUMENTATION` cmake option to it does not even bother to search for doxygen and display a warning.

**Comment by jcarpent** [comment]
Solved via #177

**Comment by ahoarau** [comment]
Awesome, thanks !
